src/common/ad_mem.c

```diff
@@ -171,9 +221,19 @@ blob_init(struct ad_blob *blob)
 	D_INIT_LIST_HEAD(&blob->bb_pgs_ckpt);
 	D_INIT_LIST_HEAD(&blob->bb_pgs_extern);
 
+	rc = d_binheap_create_inplace(DBH_FT_NOLOCK, 0, NULL, &heap_ops,
+				      &blob->bb_arena_free_heap);
+	if (rc != 0)
+		return rc;
+
 	D_ALLOC(blob->bb_pages, blob->bb_pgs_nr * sizeof(*blob->bb_pages));
 	if (!blob->bb_pages)
 		goto failed;
+	D_ALLOC(blob->bb_mh_nodes, blob->bb_pgs_nr * sizeof(struct ad_maxheap_node));
+	if (!blob->bb_mh_nodes)
+		goto failed;
+	for (i = 0; i < blob->bb_pgs_nr; i++)
+		blob->bb_mh_nodes[i].mh_active = 1;
```

Comment:
mh_active need be set active so that it could be used for allocating space, and it will set to 0 if allocation failed later, and when its free space reach to 1/4, it will be set to 1 again for allocation.

Reply:
I rename it to mh_inactive, so we don't have to init it here.